ebean-querybean/src/main/java/io/ebean/typequery/PBaseComparable.java

```diff
@@ -197,8 +198,8 @@ public final R inRange(T lower, T upper) {
    * The most common use of this could be called "effective dating" where 2 date or
    * timestamp columns represent the date range in which
    */
-  public final R inRangeWith(TQProperty<R> highProperty, T value) {
```

Comment:
I wonder, if this is correct.
Before, you could pass only properties from the model R (but with any type)
Now you can pass any property, as long as the type matches

Reply:
I believe this is better here to allow any property as long as the type matches. This specifically allows that case that Jens had where the range check was across 2 different types via a ManyToOne path (but any path should be fine).